packages/terra-form-input/src/FormInput.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './FormInput.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The defaultValue of the input field. Use this to create an uncontrolled input.
+   */
+  defaultValue: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+  /**
+   * Whether the input is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Whether the input is invalid
+   */
+  isInvalid: PropTypes.bool,
+  /**
+   * Function to trigger when the input loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Function to trigger when user changes the input value. Provide a function to create a controlled input.
+   */
+  onChange: PropTypes.func,
+  /**
+   * Function to trigger when the input gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Content to be displayed as the name.
+   */
+  name: PropTypes.string,
+  /**
+   * Whether the input is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the input field. Use this to create a controlled input.
+   */
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+};
+
+const defaultProps = {
+  defaultValue: undefined,
+  disabled: false,
+  isInvalid: false,
+  onBlur: undefined,
+  onChange: undefined,
+  onFocus: undefined,
+  name: null,
+  required: false,
+  value: undefined,
+};
+
+class FormInput extends React.Component {
+  render() {
+    const {
+      defaultValue,
+      disabled,
+      isInvalid,
+      onBlur,
+      onChange,
+      onFocus,
+      name,
+      required,
+      value,
+      ...customProps
+    } = this.props;
+
+    const attributes = Object.assign({}, customProps);
+    const formInputClassNames = cx([
+      'form-input',
+      { 'form-error': isInvalid },
+      attributes.className,
+    ]);
+
+    if (required) {
+      attributes['aria-required'] = 'true';
+    }
+
+    if (value !== undefined) {
+      attributes.value = value;
+    } else {
+      attributes.defaultValue = defaultValue;
+    }
+
+    return (
+      <input
+        aria-label="input label"
+        name={name}
+        onBlur={onBlur}
+        onChange={onChange}
+        onFocus={onFocus}
+        ref={(input) => { this.textInput = input; }}
```

Comment:
This ref should probably be removed if it's not referenced anywhere.

Reply:
Let's leave this ref. While it may not be used in this file, I could see it being useful when this component is composed into other components like time input, date picker, etc. It also lines up with how we've coded the textarea component.

Reply:
If we keep this component as a class we should be able to remove the hardcoded ref. After removing this line a consumer will be able to do the following.
```
<Input ref={this.mockFunction} />
```

Reply:
Removed. 05932d987a54c84ffd256b0c41d8925a168b67a6